### PR TITLE
[query] Fix bug with parsing Graphite find query

### PR DESCRIPTION
**Pull request by arnikola**
**What this PR does / why we need it**:

This fixes an issue with parsing graphite find queries. Previously, `foo.*` would get split up into `foo`, and `*`, which would go through a graphite globber and become `[^\.]*`, which would then not get correctly identified as a `NewFieldQuery` in the storage/index (this would convert regex values looking for `.*` only to a `NewFieldQuery`.

This PR adds `MatchField` and `MatchNotField` match types, and explicitly makes use of them in graphite queries, avoiding relying on implicit conversion in the storage layer.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:
```release-note
NONE
```

```documentation-note
NONE
```

### CODE REVIEW

**Comment by robskillington** [comment]
Ah nice, I like the approach of using explicit match types too.  Good stuff!

File: src/query/graphite/storage/converter_test.go [hunk 1]
```diff
@@ -45,6 +45,20 @@ func TestConvertMetricPartToMatcher(t *testing.T) {
 	}
 }
 
+func TestConverWildcardToMatcher(t *testing.T) {
+	metric := "*"
+	for i := 0; i < 100; i++ {
```
Comment (benraskin92) [inline src/query/graphite/storage/converter_test.go]: This is kind of a weird test -- why not just run this once?

File: src/query/models/matcher.go [hunk 2]
```diff
@@ -38,6 +38,10 @@ func (m MatchType) String() string {
 		return "=~"
 	case MatchNotRegexp:
 		return "!~"
+	case MatchField:
+		return "-"
```
Comment (benraskin92) [inline src/query/models/matcher.go]: For my understanding, how did you come up with these mappings?

File: src/query/storage/index.go [hunk 3]
```diff
@@ -192,6 +192,17 @@ func matcherToQuery(matcher models.Matcher) (idx.Query, error) {
 		}
 		return query, nil
 
+	case models.MatchNotField:
+		negate = true
+		fallthrough
```
Comment (benraskin92) [inline src/query/storage/index.go]: Nice - good use of fallthrough.

File: src/query/graphite/storage/converter_test.go (hunk 1 above)
Comment (arnikola) [reply src/query/graphite/storage/converter_test.go]: Just to ensure that the metric name is also converted to expected (i.e. `__g100__`) as well

File: src/query/models/matcher.go (hunk 2 above)
Comment (arnikola) [reply src/query/models/matcher.go]: Made 'em up :p

The only thing this affects is the `String()` method on MatchTypes

File: src/query/storage/index.go (hunk 3 above)
Comment (arnikola) [reply src/query/storage/index.go]: I learn from the best ;)

**Pull request merged**